src/peft/tuners/lora/layer.py

```diff
@@ -797,6 +838,38 @@ def get_delta_weight(self, adapter) -> torch.Tensor:
 
         return output_tensor
 
+    def _get_weight_norm(self, weight, lora_weight, scaling) -> torch.Tensor:
+        # calculate L2 norm of weight matrix, channel-wise
+        weight = weight + scaling * lora_weight
+        weight_norm = weight.norm(p=2, dim=(1, 2, 3))
+        weight_norm = weight_norm.reshape((1, self.get_base_layer().out_channels, 1, 1))
```

Comment:
Would this work instead:

`weight_norm = weight.norm(p=2, dim=(1, 2, 3), keepdim=True).transpose(1, 0)`

Reply:
Could work but I think the code is slightly more readable in the current form.

Reply:
We can leave the reshape (though view is mostly used otherwise), but I think we should avoid the call to `self.get_base_layer().out_channels`, as we have all the information we need without it. Maybe a comment that documents that shapes could be added instead.

Reply:
Fixed in 641c5b5.